File: app/src/main/java/com/jsh/kr/alltest/model/AppDataUtil.java
package com.jsh.kr.alltest.model;

import com.jsh.kr.alltest.AllTestApplication;
import com.jsh.kr.alltest.C;
import com.jsh.kr.alltestlib.util.RandomString;

public class AppDataUtil {

    private PreferenceModel preferenceModel;

    /**
     * SDMAppDataUtil create new Instance.
     */
    private static final AppDataUtil ourInstance = new AppDataUtil();

    /**
     * Gets instance.
     *
     * @return the instance
     */
    public static AppDataUtil getInstance() {
        return ourInstance;
    }

    /**
     * demo mode.
     */
    private String DEMO_MODE = "DemoMode";

    /**
     * log in response.
     */
    private String LOGINRESPONSE = "loginResponse";

    private boolean isDemoMode = false;
    private String secKey = null;
    private int isWatch = 0;

    private boolean isLoginMode = false;
    private boolean isMaintainTestMode = true;
    private int maintainTestMenu = C.Maintain.Menu_UI;
    private String maintainTestAct = null;

    /**
     * Instantiates a new Sdm app data util.
     */
    private AppDataUtil() {
        testInit();
    }

    private void testInit() {
        preferenceModel = new PreferenceModel();
        isDemoMode = false;
        secKey = null;
        isWatch = 0;
        loadData();
    }

    public void setLoginMode(boolean isLoginMode) {
        this.isLoginMode  = isLoginMode;

        preferenceModel.setLoginMode(isLoginMode);
    }

    public boolean isLoginMode() {
        return isLoginMode;
    }

    public void setMainTainTestMode(boolean isMaintainTestMode) {
        this.isMaintainTestMode = isMaintainTestMode;

        preferenceModel.setMaintainTestMode(isMaintainTestMode);
    }

    public boolean isMaintainTestMode() {
        return isMaintainTestMode;
    }

    public void setMaintainTestMenu(int maintainTestMenu) {
        this.maintainTestMenu = maintainTestMenu;

        preferenceModel.setMaintainTestMenu(maintainTestMenu);
    }

    public int getMaintainTestMenu() {
        return maintainTestMenu;
    }

    public void setMaintainTestAct(String maintainTestAct) {
        this.maintainTestAct = maintainTestAct;

        preferenceModel.setMaintainTestAct(maintainTestAct);
    }

    public String getMaintainTestAct() {
        return maintainTestAct;
    }

    private void loadData() {
        AppInfoData appInfoData = preferenceModel.getAppData();

        isLoginMode = appInfoData.isLoginMode();
        isMaintainTestMode = appInfoData.isMaintainTestMode();
        maintainTestMenu = appInfoData.getMaintainTestMenu();
        maintainTestAct = appInfoData.getMaintainTestAct();
    }

    /**
     * Sets demo mode.
     *
     * @param isDemo the is demo
     */
    public void setDemoMode(final boolean isDemo) {
//        Hawk.put(DEMO_MODE, isDemo);
        isDemoMode = isDemo;
    }

    /**
     * Is demo mode boolean.
     *
     * @return the boolean
     */
    public boolean isDemoMode() {
//        if (Hawk.isBuilt()) {
//            return Hawk.get(DEMO_MODE, false);
//        } else {
//            return false;
//        }
        return isDemoMode;
    }

    public void deleteAll(){
//        if(Hawk.isBuilt()){
//            Hawk.deleteAll();
//        }
    }

    public int getWatch() {
//        if (Hawk.isBuilt()) {
//            isWatch = Hawk.get(WATCHSTATUS + getAccountInfo().getMtsNo(), 2);
//        } else {
//            isWatch = 2;
//        }

        return isWatch;
    }

    public void setWatch(int isWatch) {
        this.isWatch = isWatch;
//        if (Hawk.isBuilt()) {
//            Hawk.put(WATCHSTATUS + getAccountInfo().getMtsNo(), isWatch);
//        }
    }

    public String getSecKey(){
        String key = null;
//        if (Hawk.isBuilt()) {
//            key = Hawk.get(SECKEY, null);
//
//            if(key == null) {
//                if(secKey == null) {
//                    secKey = createAppKey();
//                }
//                Hawk.put(SECKEY, secKey);
//            }else {
//                secKey = key;
//            }
//        } else {
//            if(secKey == null) {
//                secKey = createAppKey();
//            }
//
//        }
        if(secKey == null) {
            secKey = createAppKey();
        }
        return secKey;
    }

    private String createAppKey() {
        // AES256에 사용될 암호 키 생성

        RandomString randomString = new RandomString(32);
        String seccKey = randomString.nextString();

        return seccKey;
    }

    public void notifyWatchLoginChecking() {
//        try {
//            if (AllTestApplication.getContext() != null) {
//                CellResponse message = new CellResponse();
//                message.setPayLoad(WConstants.LOG);
//                message.setLogOutput(WConstants.MOBILETYPE, WConstants.LOG_IN_CHECKING, AppDataUtil.getInstance().getWatch());
//                WMessageSender messageSender = new WMessageSender(AllTestApplication.getContext());
//                messageSender.sendMessage(message.getOutput().toString().getBytes("UTF-8"));
//            }
//        } catch (JSONException e) {
//            LogUtil.e("SDMAppDataUtil", "JsonException", e);
//        } catch (UnsupportedEncodingException e) {
//            e.printStackTrace();
//        } catch (Exception e) {
//            LogUtil.e("SDMAppDataUtil", "notifyWatchLoginChecking", e);
//        }
    }

    /**
     * Logout.
     */
    public void notifyWatchNotLogin() {
//        try {
//            CellResponse message = new CellResponse();
//            message.setPayLoad(WConstants.LOG);
//            message.setLogOutput(WConstants.MOBILETYPE, WConstants.NOT_LOG_IN, AppDataUtil.getInstance().getWatch());
//            WMessageSender messageSender = new WMessageSender(AllTestApplication.getContext());
//            messageSender.sendMessage(message.getOutput().toString().getBytes("UTF-8"));
//        } catch (JSONException e) {
//            LogUtil.e("SDMAppDataUtil", "JsonException", e);
//        } catch (UnsupportedEncodingException e) {
//            e.printStackTrace();
//        }
    }

    /**
     * Logout.
     */
    public void logout() {
//        AllTestApplication.setLoginSuccess(false);
//        try {
//            CellResponse message = new CellResponse();
//            message.setPayLoad(WConstants.LOG_OFF);
//            message.setOutput(WConstants.MOBILETYPE);
//            WMessageSender messageSender = new WMessageSender(AllTestApplication.getContext());
//            messageSender.sendMessage(message.getOutput().toString().getBytes("UTF-8"));
//        } catch (JSONException e) {
//            LogUtil.e("SDMAppDataUtil", "JsonException", e);
//        } catch (UnsupportedEncodingException e) {
//            e.printStackTrace();
//        }
    }

    /**
     * logout without close auto login.
     */
    public void difLogOut() {
//        AllTestApplication.setLoginSuccess(false);
//        try {
//            CellResponse message = new CellResponse();
//            message.setPayLoad(WConstants.LOG_OFF);
//            message.setOutput(WConstants.MOBILETYPE);
//            WMessageSender messageSender = new WMessageSender(AllTestApplication.getContext());
//            messageSender.sendMessage(message.getOutput().toString().getBytes("UTF-8"));
//        } catch (JSONException e) {
//            LogUtil.e("SDMAppDataUtil", "JsonException", e);
//        } catch (UnsupportedEncodingException e) {
//            e.printStackTrace();
//        }
    }

    /**
     * Reset.
     */
    public void reset() {
        if (!isDemoMode()) {
            logout();
            AllTestApplication.setLoginSuccess(false);
            deleteAll();
        }
    }
}

File: appkt/src/main/java/com/jsh/kr/alltestkt/util/AppDataUtil.kt
package com.jsh.kr.alltestkt.util

import android.content.Context
import com.jsh.kr.alltestkt.AllTestApplication
import com.jsh.kr.alltestkt.constant.C
import org.json.JSONException
import java.io.UnsupportedEncodingException

/**
 * Instantiates a new Sdm app data util.
 */
object AppDataUtil {

    private const val prefName = "AppData"

    /**
     * demo mode.
     */
    private const val DEMO_MODE = "DemoMode"

    /**
     * log in response.
     */
    private const val LOGINRESPONSE = "loginResponse"

    /**
     * when run app, show login activity
     */
    private const val MODE_LOGIN = "loginMode"

    /**
     * move before test mode
     */
    private const val MODE_MAINTAIN_TEST = "MaintainTestMode"

    private const val MAINTAIN_TEST_MENU = "maintainTestDataMenu"
    private const val MAINTAIN_TEST_ACT = "maintainTestDataAct"

    /**
     * Is demo mode boolean.
     *
     * @return the boolean
     */
    /**
     * Sets demo mode.
     *
     * @param isDemo the is demo
     */
    private var isDemoMode = false
    private var secKey: String? = null
    private var watch = 0

    var isLoginMode: Boolean = false
        set(value) {
            val pref = AllTestApplication.application!!.getSharedPreferences(prefName, Context.MODE_PRIVATE)
            val editor = pref.edit()

            editor.putBoolean(MODE_LOGIN, value)

            editor.apply()
            field = value
        }

    var isMaintainTestMode = true
        set(value) {
            val pref = AllTestApplication.application!!.getSharedPreferences(prefName, Context.MODE_PRIVATE)
            val editor = pref.edit()

            editor.putBoolean(MODE_MAINTAIN_TEST, value)

            editor.apply()
            field = value
        }
    var maintainTestMenu = C.Maintain.Menu_UI
        set(value) {
            val pref = AllTestApplication.application!!.getSharedPreferences(prefName, Context.MODE_PRIVATE)
            val editor = pref.edit()

            editor.putInt(MAINTAIN_TEST_MENU, value)

            editor.apply()
            field = value
        }

    var maintainTestAct: String? = null
        set(value) {
            val pref = AllTestApplication.application!!.getSharedPreferences(prefName, Context.MODE_PRIVATE)
            val editor = pref.edit()

            editor.putString(MAINTAIN_TEST_ACT, value)

            editor.apply()
            field = value
        }

    init {
        testInit()
    }

    private fun testInit() {
        LogUtil.d(prefName, "testInit")
        isDemoMode = false
        secKey = null
        watch = 0
        loadData()
    }

    private fun loadData() {
        val pref = AllTestApplication.application!!.getSharedPreferences(prefName, Context.MODE_PRIVATE)

        isLoginMode = pref.getBoolean(MODE_LOGIN, false)
        isMaintainTestMode = pref.getBoolean(MODE_MAINTAIN_TEST, true)
        maintainTestMenu = pref.getInt(MAINTAIN_TEST_MENU, C.Maintain.Menu_UI)
        maintainTestAct = pref.getString(MAINTAIN_TEST_ACT, null)

    }

}
